print-bgp.c

```diff
@@ -2679,15 +2679,16 @@ bgp_notification_print(netdissect_options *ndo,
 	     * draft-ietf-idr-shutdown describes a method to send a communication
 	     * intended for human consumption regarding the Administrative Shutdown
 	     */
-	    if((bgpn.bgpn_minor == BGP_NOTIFY_MINOR_CEASE_SHUT ||
+	    if ((bgpn.bgpn_minor == BGP_NOTIFY_MINOR_CEASE_SHUT ||
 		bgpn.bgpn_minor == BGP_NOTIFY_MINOR_CEASE_RESET) &&
 		length >= BGP_NOTIFICATION_SIZE + 1) {
 		    tptr = dat + BGP_NOTIFICATION_SIZE;
 		    ND_TCHECK2(*tptr, 1);
 		    shutdown_comm_length = *(tptr);
 		    remainder_offset = 0;
 		    /* garbage, hexdump it all */
-		    if (shutdown_comm_length > BGP_NOTIFY_MINOR_CEASE_ADMIN_SHUTDOWN_LEN) {
+		    if (shutdown_comm_length > BGP_NOTIFY_MINOR_CEASE_ADMIN_SHUTDOWN_LEN ||
+			shutdown_comm_length > (length - BGP_NOTIFICATION_SIZE) + 1) {
```

Comment:
`>` is correct, it is legal for trailing data to exist after a shutdown communication